### ISSUE TITLE
harden(inference): tiny finite temperature overflows inv_temp to inf, collapses sampling to wrong token

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#316]
> _Issue authored by Claude (Anthropic agent) on behalf of @ohdearquant._

## Summary

A finite, positive, but extremely small `temperature` (e.g. `1e-45`) is not routed to the greedy fast path, so `inv_temp = 1.0 / temperature` overflows f32 to `+inf`. The fused temperature scaling then multiplies every logit by `inf`, collapsing all positive logits to `+inf` and destroying their ordering. The result is a degenerate distribution that returns the wrong token. Found via a discovery pass over `crates/inference/src/sampling.rs`.

## Location

`crates/inference/src/sampling.rs`:
- greedy guard at `Sampler::sample` (lines ~282-285) — catches `!temperature.is_finite()`, `temperature <= 0.0`, `top_k == 1`, but **not** a finite-tiny temperature whose reciprocal overflows
- `inv_temp = 1.0 / self.config.temperature` (lines ~332-336)
- the non-finite-max guard at `sample_top_p_with_scratch` (line ~153) papers over it by returning `candidates[0]`, but the ordering is already lost so it returns the wrong token

## Root cause

`f32::MAX ≈ 3.4028e38`, so `1.0 / temperature` overflows to `+inf` for any `temperature < ~2.94e-39` (all subnormals and the smallest normals). The existing `#295` hardening routed `1.0/inf = 0` and `1.0/NaN = NaN` temperatures to greedy, but missed the symmetric case where the temperature itself is finite-positive yet small enough that the *reciprocal* overflows.

## Reproduction

```rust
let mut s = Sampler::new(SamplingConfig {
    temperature: 1e-45, top_k: 0, top_p: 1.0, repetition_penalty: 1.0, ..
});
let tok = s.sample(&[10.0, 11.0, 9.0]);
// tok == 0 (degenerate), should be 1 (argmax — the T→0+ limit of softmax(logits/T))
```

## Severity

Latent / moderate. Not reachable from normal generation configs, but it is a real correctness gap: for `T → 0+`, `softmax(logits/T) → argmax(logits)`, so a tiny temperature must behave like greedy (return the highest-logit token), not collapse to token 0. Same family as `#225` (top_k=0, non-finite logits) and `#295` (non-finite temperature → greedy).

## Fix

Extend the greedy guard to also trigger when `1.0 / temperature` is non-finite (i.e. the reciprocal overflows). This routes tiny-temperature to the existing deterministic argmax path, which is exactly the `T→0+` limit. Byte-identical for all normal temperatures (their reciprocal is finite). Red/green test: assert `sample([10,11,9])` with `temperature=1e-45` returns token 1, not 0.
